staging/src/k8s.io/component-base/cli/flag/map_string_string.go

```diff
@@ -23,31 +23,25 @@ import (
 )
 
 // MapStringString can be set from the command line with the format `--flag "string=string"`.
-// Multiple flag invocations are supported. For example: `--flag "a=foo" --flag "b=bar"`. If this is desired
-// to be the only type invocation `NoSplit` should be set to true.
-// Multiple comma-separated key-value pairs in a single invocation are supported if `NoSplit`
-// is set to false. For example: `--flag "a=foo,b=bar"`.
+// Multiple flag invocations are supported.
+// For example: `--flag "a=foo" --flag "b=bar"`.
+// Multiple comma-separated key-value pairs in a single invocation are supported.
+// For example: `--flag "a=foo,b=bar"`.
+// If flag values may contain commas, it can be convenient to allow them unescaped
+// by passing piecemeal=true to NewMapStringString, which will cause the parser
+// to treat each invocation as a single key=value pair.
 type MapStringString struct {
 	Map         *map[string]string
 	initialized bool
-	NoSplit     bool
+	piecemeal   bool
 }
 
 // NewMapStringString takes a pointer to a map[string]string and returns the
 // MapStringString flag parsing shim for that map
-func NewMapStringString(m *map[string]string) *MapStringString {
+func NewMapStringString(m *map[string]string, piecemeal bool) *MapStringString {
```

Comment:
What is piecemeal? Needs docs in the constructor.

"piecemeal" is a strange word. I as a non-native speaker have no idea what it is. How about "commaSyntax"?

Also not that this breaks all callers. Can we add a second constructor instead? NewSingleKeyValueMapStringString

Reply:
Or `NewMapStringString(...).DisallowCommaSyntax()`

Reply:
+1 on using `commaSyntax` or `commaSplit`

Reply:
Oh oops, I think I actually meant to write [piecewise](https://en.wikipedia.org/wiki/Piecewise), which has a common math usage, not piecemeal which is idiomatic to English. But given their similarity I'm guessing piecewise isn't much better for non-native speakers?

I like the idea of a method to toggle the option, while keeping the constructor simple. Maybe `NewMapStringString(...).WithOptions(&MapStringStringOptions{foo: true})` so we don't have to add more than one method.

I'd like to find a high level word or easy to read phrase that that indicates you specify one key-value pair per flag invocation, rather that naming the option after the low-level behavior of how commas are parsed.

Reply:
+1 to using a `NewMapStringString(...).DisallowCommaSyntax()` like @sttts suggests since that way we don't change the signature of the `NewMapStringString` function (and brak backward compatibility with all usages).

Reply:
Renamed the option to `DisableBatch` and added the `WithOptions` setter. WDYT?